### PR TITLE
8T integration tests

### PR DESCRIPTION
**Pull request by vuqtran88**
### Description

This PR adds initial integration tests for the Agent Infinite Tracing functionality. The scope of this PR is limited to Infinite Tracing smoke tests on different flavors of .NET Frameworks and Core. The set of tests doesn't test all infinite tracing case scenarios.

In order to run the tests, trace observer url need to be configured in the local secrets.json. Adds the line at the root node of your local secret file.
```
{
...
...
"TraceObserverUrl" : "_REPLACE_WITH_A_TRACE_OBSERVER_URL_"
}
```

### CODE REVIEW

File: tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                }
+            );
+
+            _fixture.Initialize();
+        }
+
+        [Fact]
+        public void Test()
+        {
+            var expectedSeenCount = 2;
```
Comment (nr-ahemsath) [inline tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs:47]: Given that the way the test application is being exercised (one method that creates a transaction is being called only once), what is causing the expected count of span(s) seen, sent and received to be 2?

Comment (vuqtran88) [reply tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs:47]: https://github.com/newrelic/newrelic-dotnet-agent/blob/5b375b3c38517c8326aac0a70a4901f99f1a483c/src/Agent/NewRelic/Agent/Core/Transformers/TransactionTransformer/TransactionTransformer.cs#L322. According to this logic, we always send an extra count. I believe 1 span is the instrumented method, the extra is for root span.

Comment (nr-ahemsath) [reply tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs:47]: Cool, thanks.  I think we should add a comment to the test to explain this to the next engineer who looks at it.

Comment (vuqtran88) [reply tests/Agent/IntegrationTests/IntegrationTests/InfiniteTracing/InfiniteTracingTests.cs:48]: Added.

**Review by nr-ahemsath — APPROVED** [review]
LGTM.  New tests pass for me locally.  I added the new `TraceObserverUrl` entry to the `example-secrets.json` file.  👍

**Comment by vuqtran88** [comment]
> LGTM. New tests pass for me locally. I added the new `TraceObserverUrl` entry to the `example-secrets.json` file. 👍

Good deal !

**Pull request merged**